bb/src/tools/build.clj

```diff
@@ -0,0 +1,76 @@
+  (when (fs/exists? (:target-dir native))
+    (println (str "Cleaning up native target directory '" (:target-dir native) "'..."))
+    (fs/delete-tree (:target-dir native)))
+  (println "Done."))
+
+(defn compile
+  ([] (compile (read-edn-file "config.edn")))
+  ([{:build/keys [deps-file class-dir java-src-dirs] :as _config}]
+   (print (str "Compiling Java classes saving them to '" class-dir "'..."))
+   (b/javac {:src-dirs java-src-dirs
+             :class-dir class-dir
+             :basis (b/create-basis {:project deps-file})
+             :javac-opts ["-source" "8" "-target" "8"]})
+   (println "Done.")))
+
+(defn pom-path [{:build/keys [class-dir lib] :as _config}]
+  (b/pom-path {:lib lib
+               :class-dir class-dir}))
+
+#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
+(defn pom [{:build/keys [deps-file class-dir pom-template lib clj-src-dirs scm] :as config}]
+  (print (str "Creating pom file from template at '" pom-template "'..."))
+  (b/write-pom {:class-dir class-dir
+                :src-pom pom-template
+                :lib lib
+                :version (version/string config)
+                :basis (b/create-basis {:project deps-file})
+                :src-dirs clj-src-dirs
+                :scm (assoc scm :tag (version/sha))})
+  (println "Done." "Saved to" (pom-path config)))
+
+(defn jar-path [{:build/keys [target-dir lib] :as config}]
+  (str target-dir "/" (format "%s-%s.jar" (name lib) (version/string config))))
+
+#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
+(defn jar
+  "Returns paths to the jar and the pom file"
+  [{:build/keys [class-dir target-dir clj-src-dirs resource-dirs] :as config}]
+  (print (str "Packaging jar at '" target-dir "'..."))
+  (b/copy-dir {:src-dirs (concat clj-src-dirs resource-dirs)
+               :target-dir class-dir})
+  (b/jar {:class-dir class-dir
+          :jar-file (jar-path config)})
+  (println "Done."))
+
+(defn native-jar-path [{:build/keys [target-dir lib] :as config}]
+  (str target-dir "/" (format "%s-%s-native-shared-library.jar" (name lib) (version/string config))))
+
+#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
+(defn native-shared-library [{:build/keys [deps-file class-dir resource-dirs clj-src-dirs native] :as config}]
+  (clean config)
+  (compile config)
+  (b/copy-dir {:src-dirs (vec (concat resource-dirs clj-src-dirs (:source-dirs native)))
+               :target-dir class-dir})
+  (let [basis (b/create-basis {:project deps-file
+                               :aliases (:aliases native)})]
+    (b/compile-clj {:basis     basis
+                    :src-dirs  clj-src-dirs
+                    :class-dir class-dir})
+    (b/compile-clj {:basis     basis
+                    :src-dirs  (:source-dirs native)
+                    :class-dir class-dir})
+    (b/uber {:class-dir class-dir
+             :uber-file (native-jar-path config)
+             :basis     basis
+             :main      (:main native)})))
```

Comment:
Am I missing something or is there no command to build native-image?

Reply:
Do you mean the program `native-image`? This is an external program and has to be installed manually at the moment. The check for it was broken though and I fixed it now. 

Otherwise, there are the command `ni-cli` and `ni-build` in `bb.edn` that build native stuff.

Reply:
The native-image command line client so far only had a build profile entry in the `deps.edn` file and did not need its own build function yet.

Reply:
Ok, then it's just a confusion about the naming. In my understanding, the jar is usually a library. 

Anyway, ni-build executes the function previously in the top-level build.clj and `ni-cli` does exactly what `bin/build-native-image` did before, i.e. build the command line client. 

Would renaming `ni-build` to `ni-jar` help for understanding the difference? I would like to keep the task names short, because that makes it especially nice to work with.

Reply:
sure 👍

Reply:
Ok, I renamed it now.